tests/integration/api_v3_product_write.t

```diff
@@ -662,6 +662,21 @@ my $tests_ref = [
 			}
 		}',
 	},
+	# remove to categories (existing)
+	{
+		test_case => 'patch-tags-fields-remove',
+		method => 'PATCH',
+		path => '/api/v3/product/1234567890100',
+		body => '{
+			"fields" : "updated",
+			"product": { 
+				"categories_tags_remove": ["en:tea"],
+				"stores_tags_remove": ["Carrefour", "Mon Ptit magasin"],
+				"countries_tags_fr_remove": ["Italie", "en:spain"],
+				"labels_tags_remove": ["végétarien", "Something unrecognized in French"]
+			}
+		}',
+	},
```

Comment:
Really good that you add an integration test.

But it would be great to also add some unit tests (in tags.t) to test remove_tag_from_field (you product ref can be a hashmap with only needed fields) so that you can test more case: 
* regular case (removing 2 tags in a list of 4)
* removing values that are not in the list (should not change)
* empty list of tags to remove
* removing all tags

Reply:
Also add unit test case where you use a tagid and a value for a tag, etc.
It's really worth it.

Reply:
Thanks! 
I've been following the api logic and found Stéphane's tips really helpful. I'll make the necessary modifications to the code right away

Reply:
I made some changes, but I am a bit confused about the **is_deeply** and **is** methods. I think the **is** method is more appropriate for the remove_tags_from_field unit tests

Reply:
**is** is for scalar values

**is_deeply** for the others (to compare two arrays or two hashmaps etc.)